CAF Audio Hal

stefanhh0 opened the pull request:
Collected some commits to bring the new CAF audio HAL to yoshino. Not sure if the first two commits:
- ~makefiles: Enable audio through HDMI/DisplayPort~ -> has been removed since it is not supported on yoshino
-  PlatformConfig: Enable audio DSP instance identifier feature

Make sense for yoshino. Requires a review.

This PR depends on https://github.com/sonyxperiadev/local_manifests/pull/100 without it yoshino will be very silent.

Tested it on yoshino/lilac, here are my test results resp. observations:

Good:
- Built-In speakers and microphone only: Music playback, phone calls, system sounds like alarm clock and phone ringing
- Bluetooth connection to headset Sennheiser PXC 550: Music playback, phone calls, system sounds like alarm clock and phone ringing are replicated to go via Headset and the built-in speaker. However audio notifications are only sent to the Headset and not to the built in speaker.
- Audio cable connection to headset Sennheiser PXC 550: Music playback, system sounds like alarm clock and phone ringing are replicated to go via Headset and the built-in speaker. 

~Bad:~
~- Audio cable connection to headset Sennheiser PXC 550: Phonecalls neither I hear something nor headset microphone is useable then.~

Fixed with: 
https://github.com/sonyxperiadev/device-sony-lilac/pull/45
https://github.com/sonyxperiadev/device-sony-maple/pull/69
https://github.com/sonyxperiadev/device-sony-poplar/pull/42

Unknown:
- USB cable to the headset, would require me to have an USB C  (lilac) cable to Mini USB (Sennheiser PXC 550). I don't have such a cable. However I know that the Headset in general works when connected via USB to a computer.

Log for failing phone call via audio cable:
[default_kernel_logcat_phohnecall_via_audio_cable.log](https://github.com/sonyxperiadev/device-sony-yoshino/files/4827737/default_kernel_logcat_phohnecall_via_audio_cable.log)

With the old vendor it was working, see log:
[default_kernel_logcat_phohnecall_via_audio_cable_old.log](https://github.com/sonyxperiadev/device-sony-yoshino/files/4827821/default_kernel_logcat_phohnecall_via_audio_cable_old.log)

kholk commented:
There is no HDMI/DP on Yoshino, so the first commit that you've ported from Kumano is not applicable here.

Regarding your wired headset issue, we pointed out the solution in the chat, so apply it and it'll work great :D

stefanhh0 commented:
Removed the HDMI/DP on Yoshino commit and added additional PRs for fixing using the voice head-set via analog audio cable.

These four PRs should be then ready to be merged back.
